Provide ability to not set `all_config_vars` on heroku_app resource (#299)
heroku/terraform-provider-heroku: docs/index.md

  provided, it will be sourced from the `HEROKU_HEADERS` environment variable
  (if set).

* `customizations` - (Optional) Various attributes altering the behavior of certain resources.
  Only a single `customizations` block may be specified, and it supports the following arguments:

  * `set_app_all_config_vars_in_state` - (Optional) Controls whether the `heroku_app.all_config_vars` attribute

>     is set in the state file. Set to `false` if you'd like not to have non-managed config vars set in state.

Reviewer: Perhaps something like this to avoid the double-negative?

```suggestion
    is set in the state file. Set to `false` to only track managed config vars in the state.
```

It might also be helpful to include something explaining _why_ you'd want to change this behavior?

Author: 👍🏼
